api/chalicelib/core/feature_flags.py

```diff
@@ -0,0 +1,342 @@
+        "project_id": project_id,
+        "limit": data.limit,
+        "is_active": data.is_active,
+        "offset": (data.page - 1) * data.limit,
+    }
+
+    if data.user_id is not None:
+        constraints.append("feature_flags.created_by=%(user_id)s")
+
+    columns = (
+        "feature_flag_id",
+        "name",
+        "flag_key",
+        "description",
+        "flag_type",
+        "is_persist",
+        "is_active",
+        "created_at",
+        "updated_at",
+        "created_by",
+        "updated_by",
+    )
+
+    if data.query is not None and len(data.query) > 0:
+        constraints.append("(name ILIKE %(query)s)")
+        params["query"] = helper.values_for_operator(value=data.query,
+                                                     op=schemas.SearchEventOperator._contains)
+
+    sql = f"""
+        SELECT COUNT(feature_flags.feature_flag_id) OVER () AS count, {", ".join(columns)}
+        FROM feature_flags
+        WHERE {" AND ".join(constraints)}
+        ORDER BY created_at {data.order}
+        LIMIT %(limit)s OFFSET %(offset)s;
+    """
+
+    with pg_client.PostgresClient() as cur:
+        query = cur.mogrify(sql, params)
+        cur.execute(query)
+        rows = cur.fetchall()
+
+    if len(rows) == 0:
+        return {"total": 0, "list": []}
+
+    results = {"total": rows[0]["count"]}
+
+    rows = helper.list_to_camel_case(rows)
+    for row in rows:
+        row.pop("count")
+        row["createdAt"] = TimeUTC.datetime_to_timestamp(row["createdAt"])
+        row["updatedAt"] = TimeUTC.datetime_to_timestamp(row["updatedAt"])
+
+    results["list"] = rows
+    return results
+
+
+def get_feature_flag(project_id: int, feature_flag_id: int) -> Optional[Dict[str, Any]]:
+    """
+    Get a single feature flag by its ID.
+    """
+    feature_flag_columns = (
+        "feature_flag_id",
+        "name",
+        "flag_key",
+        "description",
+        "flag_type",
+        "is_persist",
+        "is_active",
+        "created_at",
+        "updated_at",
+        "created_by",
+        "updated_by",
+    )
+
+    feature_flag_sql = f"""
+        SELECT {", ".join(feature_flag_columns)}
+        FROM feature_flags
+        WHERE feature_flag_id = %(feature_flag_id)s AND project_id = %(project_id)s
+        AND deleted_at IS NULL
+    """
+
+    with pg_client.PostgresClient() as cur:
+        feature_flag_query = cur.mogrify(
+            feature_flag_sql,
+            {"feature_flag_id": feature_flag_id, "project_id": project_id}
+        )
+        cur.execute(feature_flag_query)
+        feature_flag_row = cur.fetchone()
+
+        if feature_flag_row is None:
+            return {"errors": ["Feature flag not found"]}
+
+        feature_flag_row["created_at"] = TimeUTC.datetime_to_timestamp(feature_flag_row["created_at"])
+        feature_flag_row["updated_at"] = TimeUTC.datetime_to_timestamp(feature_flag_row["updated_at"])
+
+    feature_flag_row["conditions"] = get_conditions(feature_flag_id)
```

Comment:
instead of getting the conditions alone, you can get them using the same query (use `LEFT JOIN` and `jsonb_agg`)
you can follow this:
https://github.com/openreplay/openreplay/blob/529218d95777937bc04a52a1f4e57fe770b997af/api/chalicelib/core/custom_metrics.py#L453